packages/core/src/services/commits/runDocumentAtCommit.ts

```diff
@@ -1,17 +1,70 @@
-import { createChain } from '@latitude-data/compiler'
-
 import {
   Commit,
+  ErrorableEntity,
   LogSources,
+  StreamType,
   type DocumentVersion,
   type Workspace,
 } from '../../browser'
 import { publisher } from '../../events/publisher'
-import { Result } from '../../lib'
-import { runChain } from '../chains/run'
+import { generateUUIDIdentifier, Result } from '../../lib'
+import { ChainResponse, runChain } from '../chains/run'
 import { createDocumentLog } from '../documentLogs'
 import { getResolvedContent } from '../documents'
 import { buildProvidersMap } from '../providerApiKeys/buildMap'
+import { RunDocumentChecker } from './RunDocumentChecker'
+
+export async function createDocumentRunResult({
+  workspace,
+  document,
+  commit,
+  errorableUuid,
+  parameters,
+  resolvedContent,
+  source,
+  response,
+  duration,
+}: {
+  workspace: Workspace
+  commit: Commit
+  document: DocumentVersion
+  source: LogSources
+  errorableUuid: string
+  parameters: Record<string, unknown>
+  resolvedContent: string
+  customIdentifier?: string
+  duration?: number
+  response?: ChainResponse<StreamType>
+}) {
+  const durantionInMs = duration ?? 0
+  publisher.publishLater({
```

Comment:
hmm you are publish the documentrun event even if the document run failed, is that ok? any event handlers that we should be worried about?

Reply:
Talked with Cesar. Not publishing if there is error